inception-kb/src/main/java/de/tudarmstadt/ukp/inception/kb/exporter/KnowledgeBaseExporter.java

```diff
@@ -0,0 +1,193 @@
+import org.eclipse.rdf4j.repository.sparql.config.SPARQLRepositoryConfig;
+import org.eclipse.rdf4j.rio.RDFFormat;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import de.tudarmstadt.ukp.clarin.webanno.api.export.ProjectExportRequest;
+import de.tudarmstadt.ukp.clarin.webanno.api.export.ProjectExporter;
+import de.tudarmstadt.ukp.clarin.webanno.api.export.ProjectImportRequest;
+import de.tudarmstadt.ukp.clarin.webanno.export.model.ExportedProject;
+import de.tudarmstadt.ukp.clarin.webanno.model.Project;
+import de.tudarmstadt.ukp.inception.kb.KnowledgeBaseService;
+import de.tudarmstadt.ukp.inception.kb.RepositoryType;
+import de.tudarmstadt.ukp.inception.kb.model.KnowledgeBase;
+import de.tudarmstadt.ukp.inception.kb.reification.Reification;
+
+@Component
+public class KnowledgeBaseExporter implements ProjectExporter
+{
+    private static final String KEY = "knowledgeBases";
+    private static final Logger LOG = LoggerFactory.getLogger(KnowledgeBaseExporter.class);
+    
+    private static final String SOURCE = "knowledgeBase_source";
+    private static final String SOURCE_FOLDER = "/" + SOURCE;
+    private static final String KNOWLEDGEBASEFILES = SOURCE_FOLDER + "/";
+    
+    private static final RDFFormat knowledgeBaseFileExportFormat = RDFFormat.TURTLE;
+    
+    private final KnowledgeBaseService kbService;
+    
+    @Autowired
+    public KnowledgeBaseExporter(KnowledgeBaseService knowledgeBaseService)
+    {
+        kbService = knowledgeBaseService;
+    }
+    
+    @Override
+    public void exportData(ProjectExportRequest aRequest, ExportedProject aExProject, File aFile)
+        throws Exception
+    {
+        Project project = aRequest.getProject();
+        List<ExportedKnowledgeBase> exportedKnowledgeBases = new ArrayList<>();
+        for (KnowledgeBase kb: kbService.getKnowledgeBases(project)) {
+            ExportedKnowledgeBase exportedKB = new ExportedKnowledgeBase();
+            exportedKB.setName(kb.getName());
+            exportedKB.setType(kb.getType().toString());
+            exportedKB.setClassIri(kb.getClassIri().stringValue());
+            exportedKB.setSubclassIri(kb.getSubclassIri().stringValue());
+            exportedKB.setTypeIri(kb.getTypeIri().stringValue());
+            exportedKB.setDescriptionIri(kb.getDescriptionIri().stringValue());
+            exportedKB.setLabelIri(kb.getLabelIri().stringValue());
+            exportedKB.setPropertyTypeIri(kb.getPropertyTypeIri().stringValue());
+            exportedKB.setReadOnly(kb.isReadOnly());
+            exportedKB.setEnabled(kb.isEnabled());
+            exportedKB.setReification(kb.getReification().toString());
+            exportedKB.setSupportConceptLinking(kb.isSupportConceptLinking());
+            exportedKB.setBasePrefix(kb.getBasePrefix());
+            exportedKnowledgeBases.add(exportedKB);
```

Comment:
@reckart Do we need to export anything else apart from these? Are we looking for the changes made in KB as well?

Reply:
Basically, we need all the properties of the `KnowledgeBase` class so that we can re-create it from the JSON DTO. What do you mean by "the changes made in the KB"?